src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,612 @@
+        logging.debug("Executing: %s" % url)
+        if method == _METHOD_GET:
+            resp = requests.get(url, json=data, timeout=timeout)
+        elif method == _METHOD_POST:
+            resp = requests.post(url, json=data, timeout=timeout)
+
+        if resp.status_code != expected_status:
+            raise AsmApiException(url, resp, expected_status)
+
+        logging.debug("Call to %s went fine, no problems occured\n" % url)
+
+        if raw_response:
+            return resp
+        elif method == _METHOD_POST:
+            return resp.status_code
+        elif method == _METHOD_GET:
+            return json.loads(resp.content)
+
+
+class EBeamScanner(model.Emitter):
+
+    def __init__(self, name, role, parent, **kwargs):
+        super(EBeamScanner, self).__init__(name, role, **kwargs)
+        self.parent = parent
+
+        # self._metadata[model.MD_PIXEL_SIZE] =
+
+        clockFrequency = self.parent.ASMAPICall(self.parent.server_url + "/scan/clock_frequency", _METHOD_GET, 200)[
+            'frequency']
+        self.clockPeriod = model.FloatVA(1 / clockFrequency, unit='s')
+        self._shape = model.TupleVA((6400, 6400), unit='px')
+        # The resolution min/maximum are derived from the effective cell size restriction defined in the API
+        self.resolution = model.ResolutionVA((6400, 6400), ((10, 10), (1000 * 8, 1000 * 8)))
+        self.dwellTime = model.FloatContinuous(self.clockPeriod.value, (max(self.clockPeriod.value, 400e-9), 100.0),
+                                               unit='s')
+        self.pixelSize = model.TupleContinuous((4, 4), range=((1, 1), (100000, 100000)), unit='m',
+                                               setter=self._setPixelSize)
+        self.rotation = model.FloatContinuous(0, range=(0, 2 * numpy.pi), unit='rad')
+        self.scanFlyback = model.FloatVA(0, unit='s')
+        self.scanOffset = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V')
+        self.scanGain = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V')
+        self.scanDelay = model.TupleContinuous((0, 0), range=((0, 0), (100000, 100000)), unit='s',
+                                               setter=self._setScanDelay)
+
+        self._metadata[model.MD_PIXEL_SIZE] = self.pixelSize.value
+        self._metadata[model.MD_DWELL_TIME] = self.dwellTime.value
+
+    def _setPixelSize(self, pixelSize):
+        if pixelSize[0] == pixelSize[1]:
+            return pixelSize
+        else:
+            logging.warning("Non-square pixel size entered, only square pixel sizes are supported. "
+                            "Width of pixel size is used as height.")
+            return (pixelSize[0], pixelSize[0])
+
+    def _setScanDelay(self, scanDelay):
+        # Check if detector can record images before ebeam scanner has started to scan.
+        if not (hasattr(self.parent, "_mppc")) or self.parent._mppc.acqDelay.value - scanDelay[0] >= 0:
```

Comment:
I added that check to prevent any weird error in case the object was initialized differently that it is now. For examples if you didn't create the ebeamscanner object via the ASM server or have an inhertited object from the ASM server maybe the mppc object is not created. It could even be that they are not added in the right order. That would give a weird error and this check prevents that.

Reply:
Maybe your comment on line 156 answers my comment already:
> be less defensive against your own code

Reply:
If I understand Eric correctly, he is saying, that when ever calling this method, you can safely assume that the mppc was already instantiated.
@pieleric Just putting this method as setter for a VA in the init of the scanner is no problem, while the mppc was not instantiated, correct? As scanner is before mppc instantiated.